### PR TITLE
solution

### PR DESCRIPTION
**Pull request by PyCore23**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,74 @@
...
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            return Distance((self.km + other.km))
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        return print("error")
```
Comment (pavlejviki) [inline app/main.py:20]: this method is supposed to return an instance of Distance, and not the print function

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,74 @@
...
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            return Distance((self.km + other.km))
+        elif isinstance(other, (int, float)):
```
Comment (pavlejviki) [inline app/main.py:18]: this check is redundant, if other is not an instance of Distance, than it's int or float

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,74 @@
...
+            return Distance(self.km + other)
+        return print("error")
+
+    def __iadd__(self, other: Distance | int | float) -> Distance | int | float:
```
Comment (pavlejviki) [inline app/main.py:22]: wrong return type annotation

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,74 @@
...
+        if isinstance(other, Distance):
+            self.km = self.km + other.km
+            return self
+        elif isinstance(other, (int, float)):
+            self.km = self.km + other
+            return self
```
Comment (pavlejviki) [inline app/main.py:28]: you can use just one return

File: app/main.py [hunk 5]
```diff
@@ -1,3 +1,74 @@
...
+        return print("error")
```
Comment (pavlejviki) [inline app/main.py:29]: remove the `return print ` in all your methods

File: app/main.py [hunk 6]
```diff
@@ -1,3 +1,74 @@
...
+
+    def __mul__(self, other: Distance | int | float) -> Distance | int | float:
```
Comment (pavlejviki) [inline app/main.py:31]: wrong return type annotation

File: app/main.py [hunk 7]
```diff
@@ -1,3 +1,74 @@
...
+
+    def __truediv__(self, other: Distance | int | float) -> Distance | int | float:
+        if isinstance(other, (int, float)):
+            return Distance(round((self.km / other), 2))
+        return print("error")
+
+    def __lt__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km < other.km
+        elif isinstance(other, (int, float)):
```
Comment (pavlejviki) [inline app/main.py:44]: this check is redundant

File: app/main.py [hunk 8]
```diff
@@ -1,3 +1,74 @@
...
+        return print("error")
+
+    def __eq__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        elif isinstance(other, (int, float)):
+            return self.km == other
+        return print("error")
+
+    def __le__(self, other: Distance | int | float) -> bool:
```
Comment (pavlejviki) [inline app/main.py:62]: You can use previously created magic method here(and in ge as well) to simplify your code. e.g. return not self > other

**Review by pavlejviki — CHANGES_REQUESTED** [review]

**Review by viktoria-rybenchuk — CHANGES_REQUESTED** [review]
fix code related to comments

**Review by viktoria-rybenchuk — CHANGES_REQUESTED** [review]
Rollback changes in README.md and fix all comments

**Review by Polyakiv-Andrey — APPROVED** [review]
Good job!

File: app/main.py [hunk 9]
```diff
@@ -1,3 +1,64 @@
...
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(
+            self,
+            other: Distance | int | float
+    ) -> Distance | int | float:
```
Comment (pavlejviki) [inline app/main.py:30]:
```suggestion
    ) -> Distance:
```

File: app/main.py [hunk 10]
```diff
@@ -1,3 +1,64 @@
...
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+
+    def __truediv__(
+            self,
+            other: Distance | int | float
+    ) -> Distance | int | float:
```
Comment (pavlejviki) [inline app/main.py:37]:
```suggestion
    ) -> Distance:
```

File: app/main.py [hunk 11]
```diff
@@ -1,3 +1,64 @@
...
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(
```
Comment (pavlejviki) [inline app/main.py:27]: mul and truediv methods are supposed to accept only int  and  float

**Review by pavlejviki — CHANGES_REQUESTED** [review]

**Pull request closed without merging**